izpack-core/src/test/java/com/izforge/izpack/core/data/DefaultVariablesTest.java

```diff
@@ -240,10 +246,56 @@ public void testConditionalDynamicVariables()
         assertEquals("/usr/local/bin", variables.get("INSTALL_PATH"));
     }
 
+    @Rule
+    public TemporaryFolder rootFolder = new TemporaryFolder();
+
     /**
-     * Tests simple dynamic variables.
+     * Tests for IZPACK-1260
      */
     @Test
+    public void testDynamicVariablesIZPACK1260()
+    {
+        // set up conditions
+        Map<String, Condition> conditions = new HashMap<String, Condition>();
+        conditions.put("haveInstallPath", new ExistsCondition(ExistsCondition.ContentType.VARIABLE, "INSTALL_PATH"));
+        conditions.put("previous.wrapper.conf.exists", new ExistsCondition(ExistsCondition.ContentType.FILE, "${previous.wrapper.conf}"));
+
+        // set up the rules
+        AutomatedInstallData installData = new AutomatedInstallData(variables, Platforms.LINUX);
+        RulesEngineImpl rules = new RulesEngineImpl(installData, new ConditionContainer(new DefaultContainer()),
+                installData.getPlatform());
+        rules.readConditionMap(conditions);
+        ((DefaultVariables) variables).setRules(rules);
+
+        variables.add(createDynamicCheckonce("previous.wrapper.conf", "${INSTALL_PATH}/conf/wrapper.conf", "haveInstallPath"));
+        variables.add(createDynamicCheckonce("previous.wrapper.conf", "${INSTALL_PATH}/wrapper.conf", "haveInstallPath+!previous.wrapper.conf.exists"));
+
+        File installPath = null;
+        File confFile = null;
+        try
+        {
+            installPath = rootFolder.newFolder("myapp");
+            File confPath = new File(installPath, "conf");
+            confPath.mkdirs();
+            confFile = new File(confPath, "wrapper.conf");
+            FileUtils.touch(confFile);
+        }
+        catch (IOException e)
+        {
+            fail("File operation failed.");
+        }
+
+        variables.set("INSTALL_PATH", installPath.getAbsolutePath());
+
+        variables.refresh();
+
+        assertEquals(confFile.getAbsolutePath(), variables.get("previous.wrapper.conf"));
```

Comment:
> Yes, this would be great!

Should be fixed (in theory), see above, refresh the code from this PR and let me know whether it works for you.